lib/common/phpstan.neon.dist

```diff
@@ -2,11 +2,15 @@ includes:
 	# @see https://github.com/phpstan/phpstan-src/blob/master/conf/bleedingEdge.neon
 	- phar://phpstan.phar/conf/bleedingEdge.neon
 parameters:
-	level: 3
+	level: 4
 	inferPrivatePropertyTypeFromConstructor: true
 	paths:
 		- %currentWorkingDirectory%/src/
 	autoload_files:
 		- %currentWorkingDirectory%/vendor/autoload.php
 	ignoreErrors:
 		- '#^PHPDoc tag @throws with type AmpProject\\Exception\\FailedRemoteRequest is not subtype of Throwable$#'
+		-
+			message: '#^If condition is always false\.$#'
+			path: 'src/Dom/Document.php'
+			# See https://github.com/phpstan/phpstan/issues/3291
```

Comment:
I assume your PR on the phpstan repo (https://github.com/phpstan/phpstan-src/pull/203) fixes this. Once a new release is made with that change we should be able to remove this.

Reply:
Yes. The general idea is to update to latest PHPStan and revisit the exceptions here each time we bump the level.

Reply:
FWIW I think PHPStan will complain if an exception is no longer encountered. In other words, it will tell you that you can remove it :-)

Reply:
Yes, indeed, it does.